### PR TITLE
Custom section summaries

### PR DESCRIPTION
**Pull request by JamesGardiner**
### PR Context
Implements custom section summaries as described [here](https://github.com/ONSdigital/eq-questionnaire-validator/blob/master/doc/decisions/0006-custom-section-summaries.md).

Relevant schemas definitions and test definitions have been updated. 

### Checklist

* [ ] eq-translations updated to support any new schema keys which need translation

### CODE REVIEW

**Comment by JamesGardiner** [comment]
[Implement custom section summaries](https://trello.com/c/ZYz6mKNZ/3675-implement-custom-section-summaries)

File: schemas/blocks/section_summary.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "block": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "$ref": "https://eq.ons.gov.uk/common_definitions.json#/identifier"
+      },
+      "type": {
+        "type": "string",
+        "enum": ["SectionSummary"]
```
Comment (ajmaddaford) [inline schemas/blocks/section_summary.json]: The enum in `schemas/blocks/summary.json` needs to be updated (`SectionSummary` removed).

Comment (JamesGardiner) [reply schemas/blocks/section_summary.json]: Yep - and SectionSummary definition needed adding to the questionnaire_v1 schema. Should all be fixed in 45baaf2

File: app/validation/validator.py [hunk 2]
```diff
@@ -128,13 +128,21 @@ def validate_questionnaire(self, json_to_validate):
...
 
         if section_repeat:
             errors.extend(self._validate_list_exists(section_repeat["for_list"]))
             errors.extend(
                 self._validate_placeholder_object(section_repeat["title"], None)
             )
+
+        custom_section_summary = section.get("summary")
+        if custom_section_summary:
+            for custom_section_element in custom_section_summary.get("items", []):
```
Comment (ajmaddaford) [inline app/validation/validator.py]: The naming here needs to change. The existence of a `summary` element does not mean it's a custom section summary as the `collapsible` property is also possible. A custom section summary exists if there is an `items` property.

File: app/validation/validator.py [hunk 3]
```diff
@@ -128,13 +128,21 @@ def validate_questionnaire(self, json_to_validate):
...
-        section_repeat = section.get("repeat", {})
+        section_repeat = section.get("repeat")
 
         if section_repeat:
             errors.extend(self._validate_list_exists(section_repeat["for_list"]))
             errors.extend(
                 self._validate_placeholder_object(section_repeat["title"], None)
             )
+
+        custom_section_summary = section.get("summary")
```
Comment (LJBabbage) [inline app/validation/validator.py]: Not a fan of custom_section_summary in the code and summary in the JSON summary.

Comment (JamesGardiner) [reply app/validation/validator.py]: I went with `section_summary` as it is consistent with `section_repeat`

File: app/validation/validator.py (hunk 2 above)
Comment (JamesGardiner) [reply app/validation/validator.py]: Changed to `section_summary` and `items`

File: schemas/blocks/list_collector_driving_question.json [hunk 4]
```diff
@@ -17,9 +17,6 @@
         "type": "string",
         "enum": ["ListCollectorDrivingQuestion"]
       },
-      "show_on_section_summary": {
```
Comment (LJBabbage) [inline schemas/blocks/list_collector_driving_question.json]: Why are we removing show on section_summary?

**Pull request merged**